tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -640,6 +649,55 @@ EOF
 		chmod g+rx,o+x "${ROOTFS_DIR}"
 	fi
 
+	if [ "${AGENT_POLICY}" == "yes" ]; then
+		# Setup systemd-based environment for kata-opa.
+		local opa_bin_dir="$(get_opa_bin_dir "${ROOTFS_DIR}")"
+		if [ -z "${opa_bin_dir}" ]; then
+			# OPA was not installed already, so download it here.
+			#
+			# TODO: if an OPA package is not available for the Guest image distro,
+			#   	Kata should cache the OPA source code, toolchain information, etc.
+			#   	OPA should be built from the cached source code instead of downloading
+			#   	this binary.
+			#
+			opa_bin_url="$(get_package_version_from_kata_yaml externals.open-policy-agent.meta.binary)"
+			info "Downloading OPA binary from ${opa_bin_url}"
+			curl --fail -L "${opa_bin_url}" -o opa || die "Failed to download OPA"
+
+			# Install the OPA binary.
+			opa_bin_dir="/usr/local/bin"
+			local opa_bin="${ROOTFS_DIR}${opa_bin_dir}/opa"
+			info "Installing OPA binary to ${opa_bin}"
+			install -D -o root -g root -m 0755 opa -T "${opa_bin}"
+		else
+			info "OPA binary already exists in ${opa_bin_dir}"
+		fi
+
+		# Install default settings for the kata-opa service.
+		local kata_opa_in_dir="${script_dir}/../../../src/kata-opa"
+		local opa_settings_dir="/etc/kata-opa"
+		local policy_file="allow-all.rego"
+		local policy_dir="${ROOTFS_DIR}/${opa_settings_dir}"
+		mkdir -p "${policy_dir}"
+		install -D -o root -g root -m 0644 "${kata_opa_in_dir}/${policy_file}" -T "${policy_dir}/${policy_file}"
+		ln -sf "${policy_file}" "${policy_dir}/default-policy.rego"
```

Comment:
It seems likely that different runtime classes will want different policies, at least for people who are using host data. What is best way to support that?

Reply:
I agree. Solution: TBD, in a future patch.

Reply:
This is an orthogonal discussion, but kata-deploy can take care of specific policies for specific runtime classes.

We're already dealing (or about to deal) with specific node feature rules for specific runtime-classes, and the logic can just be followed based on what I'm doing for NFD enablement.